daml-assistant/src/DAML/Assistant/Env.hs

```diff
@@ -90,18 +130,40 @@ testDamlEnv Env{..} = firstJustM (\(test, msg) -> unlessMaybeM test (pure msg))
       , "The project directory does not exist. Please check if DAML_PROJECT is incorrectly set.")
     ]
 
+-- | Determine the absolute path to the assistant. Can be overriden with
+-- DAML_ASSISTANT env var.
+getDamlAssistantPath :: DamlPath -> IO DamlAssistantPath
+getDamlAssistantPath damlPath =
+    overrideWithEnvVar damlAssistantEnvVar DamlAssistantPath $
+        getDamlAssistantPathDefault damlPath
+
 -- | Determine the absolute path to the assistant.
-getDamlAssistantPath :: Applicative f => DamlPath -> f DamlAssistantPath
-getDamlAssistantPath (DamlPath damlPath)
+getDamlAssistantPathDefault :: Applicative f => DamlPath -> f DamlAssistantPath
+getDamlAssistantPathDefault (DamlPath damlPath)
     -- Our packaging logic for Haskell results in getExecutablePath
     -- pointing to the dynamic linker and getProgramName returning "daml" in
     -- both cases so we use this hack to figure out the executable name.
     | takeFileName damlPath == ".daml-head" = pure $ DamlAssistantPath $ damlPath </> "bin" </> "daml-head"
     | otherwise = pure $ DamlAssistantPath $ damlPath </> "bin" </> "daml"
 
--- | Determine SDK version of running daml assistant.
+-- | Determine SDK version of running daml assistant. Can be overriden
+-- with DAML_ASSISTANT_VERSION env var.
 getDamlAssistantSdkVersion :: IO (Maybe DamlAssistantSdkVersion)
-getDamlAssistantSdkVersion = fmap DamlAssistantSdkVersion <$> do
+getDamlAssistantSdkVersion = do
+    versionStrM <- getEnv damlAssistantVersionEnvVar
```

Comment:
That looks like you are replicating `overrideWithEnvVarMaybe`.

Reply:
oops!